tests/drivers/eeprom/src/main.c

```diff
@@ -43,27 +43,148 @@ static void test_write_and_verify(void)
 	const uint8_t wr_buf2[sizeof(wr_buf1)] = { 0xAA, 0xBB, 0xCC, 0xDD };
 	uint8_t rd_buf[sizeof(wr_buf1)];
 	const struct device *eeprom;
+	size_t size;
+	off_t address;
 	int rc;
 
 	eeprom = device_get_binding(eeprom_label);
+	size = eeprom_get_size(eeprom);
 
-	rc = eeprom_write(eeprom, 0, wr_buf1, sizeof(wr_buf1));
-	zassert_equal(0, rc, "Unexpected error code (%d)", rc);
+	address = 0;
+	while (address < size) {
+		rc = eeprom_write(eeprom, address, wr_buf1, sizeof(wr_buf1));
+		zassert_equal(0, rc, "Unexpected error code (%d)", rc);
 
-	rc = eeprom_read(eeprom, 0, rd_buf, sizeof(rd_buf));
-	zassert_equal(0, rc, "Unexpected error code (%d)", rc);
+		rc = eeprom_read(eeprom, address, rd_buf, sizeof(rd_buf));
+		zassert_equal(0, rc, "Unexpected error code (%d)", rc);
 
-	rc = memcmp(wr_buf1, rd_buf, sizeof(wr_buf1));
-	zassert_equal(0, rc, "Unexpected error code (%d)", rc);
+		rc = memcmp(wr_buf1, rd_buf, sizeof(wr_buf1));
+		zassert_equal(0, rc, "Unexpected error code (%d)", rc);
 
-	rc = eeprom_write(eeprom, 0, wr_buf2, sizeof(wr_buf2));
-	zassert_equal(0, rc, "Unexpected error code (%d)", rc);
+		address += sizeof(wr_buf1);
+	}
 
-	rc = eeprom_read(eeprom, 0, rd_buf, sizeof(rd_buf));
-	zassert_equal(0, rc, "Unexpected error code (%d)", rc);
+	address = 0;
+	while (address < size) {
+		rc = eeprom_write(eeprom, address, wr_buf2, sizeof(wr_buf2));
+		zassert_equal(0, rc, "Unexpected error code (%d)", rc);
```

Comment:
This seems to be writing the whole EEPROM. If so, why is that necessary for a test? It should be possible to test things with just a few writes.

Reply:
done

Reply:
Actually it is, because it does not increase test coverage, but adds more time. It is harder to debug a failing test that does way more than it needs to. The log would be enormouse!

Tests should be written to cover the functionality present.

Reply:
Ok, I will reduce it.